demos/digit-prediction/digits_prediction.nlogo

```diff
@@ -0,0 +1,527 @@
+extensions [ py ]
+
+globals [ path coords_x coords_y color_list digit]
+
+to setup-py
+  clear-all
+  py:setup py:python
+  (py:run
+    "import numpy as np"
+    "from keras.models import load_model"
+  )
+  ;; load model
+  py:run "model = load_model(\"model.h5\")"
+
+  ;; create coords list, useful to make color_list
+  set coords_x (range max-pxcor)
+  set coords_y (range max-pycor)
+  set color_list []
+end
+
+to draw
+  ifelse mouse-down?
+  [
+    ask patch mouse-xcor mouse-ycor
+    [
+      ifelse rubber
+      [
+        set pcolor black
+      ]
+      [
+        set pcolor white
+      ]
+    ]
+  ]
+  [
+    predict
+  ]
+  display
+end
+
+to predict
+  set color_list []
+  foreach coords_x
+  [
+    x -> foreach coords_y
+    [
+      y ->
+        let c [ pcolor / 10 ] of patch x y
+        set color_list lput c color_list
+    ]
+  ]
+  py:set "images" color_list
+  py:run "array = np.array(images).reshape(28, 28)"
+  py:run "array = array.reshape(1, 28, 28, 1)"
+  py:run "y_pred = model.predict(array)[0]"
+  py:run "result = np.argmax(y_pred)"
```

Comment:
It could be cool to show the probabilities themselves (I'm assuming the last layer is a softmax). You could even show the probabilities for each of the digits in different monitors. Not necessary, just could be cool.